File: 단계별로 풀어보기/조건문/5단계/알람시계.cpp
#include <iostream>
using namespace std;

int main(){
  int hour, minute;
  cin>>hour>>minute;
  
  minute=minute-45;
  
  if(minute<0){
    hour=hour-1;
    minute=60+minute;
  }
  if(hour==-1){
    hour=23;
  }
  
  cout<<hour<<' '<<minute<<endl;
  
  return 0;
}

File: 단계별로 풀어보기/조건문/5단계/알람 시계.py
def alarm():
  hour, minute =map(int,input().split(' '))
  
  minute-=45
  
  if(minute<0):
    hour-=1
    minute+=60
    
  if(hour==-1):
    hour=23
    
  print(str(hour)+' '+str(minute))

if(__name__=='__main__'):
  alarm()
